continuous_integration/travis/test_imports.sh

```diff
@@ -30,6 +30,6 @@ source activate test-imports
 (test_import "Core" "" "import dask, dask.threaded, dask.optimization") && \
 (test_import "Delayed" "toolz" "import dask.delayed") && \
 (test_import "Bag" "toolz partd cloudpickle" "import dask.bag") && \
-(test_import "Array" "toolz numpy" "import dask.array") && \
+(test_import "Array" "toolz numpy crick" "import dask.array") && \
```

Comment:
crick should not be a mandatory dependency for dask array - it could be *optional* for certain functionality (e.g. `percentile`), but should only be needed if requested.

Reply:
Removed crick from being a mandatory dependency.

Reply:
Btw is there a list of optional dependencies somewhere where I should add it?

Reply:
We should add it to the install scripts on `travis` and `appveyor`, but other than that no. 

There's also `dask.utils.import_required` which is used in a few places to provide a nice error message when importing a dependency fails for some optional functionality. I'd probably add this to `quantile` and `percentile` to error nicely when they're called by `crick` isn't installed. See e.g. https://github.com/dask/dask/blob/9f870d17e6fcae00ab010c86a5b182b2ecf214b8/dask/bag/avro.py#L95-L97

Reply:
I added it to the install scripts and also added those import warnings.